jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -106,8 +107,73 @@
+        .with(new HttpField(HttpHeader.ACCEPT, "text/html,application/xhtml+xml,application/xmlq=0.9,*/*q=0.8"))
+        .with(new HttpField(HttpHeader.ACCEPT, "text/html,application/xhtml+xml,application/xmlq=0.9,image/webp,image/apng,*/*q=0.8"))
+        .with(new HttpField(HttpHeader.ACCEPT_RANGES, HttpHeaderValue.BYTES))
+        .with(new HttpField(HttpHeader.PRAGMA, "no-cache"))
+        .with(new HttpField(HttpHeader.CACHE_CONTROL, "private, no-cache, no-cache=Set-Cookie, proxy-revalidate"))
+        .with(new HttpField(HttpHeader.CACHE_CONTROL, "no-cache"))
+        .with(new HttpField(HttpHeader.CACHE_CONTROL, "max-age=0"))
+        .with(new HttpField(HttpHeader.CONTENT_LENGTH, "0"))
+        .with(new HttpField(HttpHeader.CONTENT_ENCODING, "gzip"))
+        .with(new HttpField(HttpHeader.CONTENT_ENCODING, "deflate"))
+        .with(new HttpField(HttpHeader.TRANSFER_ENCODING, "chunked"))
+        .with(new HttpField(HttpHeader.EXPIRES, "Fri, 01 Jan 1990 00:00:00 GMT"))
+        .withAll(() ->
+        {
+            Map<String, HttpField> map = new LinkedHashMap<>();
+            // Add common Content types as fields
+            for (String type : new String[]{
+                "text/plain", "text/html", "text/xml", "text/json", "application/json", "application/x-www-form-urlencoded"
+            })
+            {
+                HttpField field = new PreEncodedHttpField(HttpHeader.CONTENT_TYPE, type);
+                map.put(field.toString(), field);
+
+                for (String charset : new String[]{"utf-8", "iso-8859-1"})
+                {
+                    PreEncodedHttpField field1 = new PreEncodedHttpField(HttpHeader.CONTENT_TYPE, type + ";charset=" + charset);
+                    map.put(field1.toString(), field1);
+                    PreEncodedHttpField field2 = new PreEncodedHttpField(HttpHeader.CONTENT_TYPE, type + "; charset=" + charset);
+                    map.put(field2.toString(), field2);
+                    PreEncodedHttpField field3 = new PreEncodedHttpField(HttpHeader.CONTENT_TYPE, type + ";charset=" + charset.toUpperCase(Locale.ENGLISH));
+                    map.put(field3.toString(), field3);
+                    PreEncodedHttpField field4 = new PreEncodedHttpField(HttpHeader.CONTENT_TYPE, type + "; charset=" + charset.toUpperCase(Locale.ENGLISH));
+                    map.put(field4.toString(), field4);
+                }
+            }
+            return map;
+        })
+        .withAll(() ->
+        {
+            Map<String, HttpField> map = new LinkedHashMap<>();
+            for (HttpHeader h : HttpHeader.values())
+            {
+                HttpField httpField = new HttpField(h, (String)null);
+                map.put(httpField.toString(), httpField);
+            }
+            return map;
+        })
+        .build();
+    private static final MutableIndex<HttpField> NO_CACHE = new Index.Builder<HttpField>()
+        .caseSensitive(false)
+        .mutable()
+        .build();
```

Comment:
This will create an unbounded shared cache!

Should be something like:
```suggestion
    private static final MutableIndex<HttpField> NO_CACHE = new Index.Builder<HttpField>()
        .caseSensitive(false)
        .mutable()
        .maxCapacity(0)
        .build();
```

Hopefully the builder will be clever when is sees a maxCapacity of 0 and return an `EmptyIndex` impl

Reply:
Good idea, let me add that in.